moveit_core/trajectory_processing/include/moveit/trajectory_processing/time_optimal_trajectory_generation.h

```diff
@@ -99,25 +111,23 @@ class Path
   std::list<std::pair<double, bool>> getSwitchingPoints() const;
 
 private:
+  // Default constructor private to prevent misuse. Use `create` instead to create a Path object.
+  Path() = default;
+
   PathSegment* getPathSegment(double& s) const;
-  double length_;
+
+  double length_ = 0.0;
   std::list<std::pair<double, bool>> switching_points_;
   std::list<std::unique_ptr<PathSegment>> path_segments_;
 };
 
 class Trajectory
 {
 public:
-  /// @brief Generates a time-optimal trajectory
-  Trajectory(const Path& path, const Eigen::VectorXd& max_velocity, const Eigen::VectorXd& max_acceleration,
-             double time_step = 0.001);
-
-  ~Trajectory();
-
-  /** @brief Call this method after constructing the object to make sure the
-     trajectory generation succeeded without errors. If this method returns
-     false, all other methods have undefined behavior. **/
-  bool isValid() const;
+  /// @brief Generates a time-optimal trajectory.
+  /// @returns std::nullopt if the trajectory couldn't be parameterized.
+  static std::optional<Trajectory> create(const Path& path, const Eigen::VectorXd& max_velocity,
+                                          const Eigen::VectorXd& max_acceleration, double time_step = 0.001);
```

Comment:
I left this as is, since it's only used here. It's convenient to see the default value right here.
The other one I made it a constant because it's used in two places.